solidity/contracts/PetalLock.sol

```diff
@@ -100,7 +101,8 @@ contract PetalLock {
 		}
 		
 		// set content hash
-		ensPublicResolver.setContenthash(pathToChild[finalChildIndex].node, contenthash);
+		if (contenthash.length != 0) { ensPublicResolver.setContenthash(pathToChild[finalChildIndex].node, contenthash); }
+		if (resolutionAddress != address(0x0)) { ensPublicResolver.setAddr(pathToChild[finalChildIndex].node, resolutionAddress); }
```

Comment:
```suggestion
		if (contenthash.length != 0) { ensPublicResolver.setContenthash(pathToChild[finalChildIndex].node, contenthash) }
		if (resolutionAddress != address(0x0)) { ensPublicResolver.setAddr(pathToChild[finalChildIndex].node, resolutionAddress) }
```

Reply:
Solidity requires ;'s

Reply:
Ah, didn't realize this was a Solidity file.  😆  This particular code block is close enough to TS that a quick scan didn't switch my brain.